docs/articles/multi-attribute-semantic-search.md

```diff
@@ -0,0 +1,459 @@
+| Thornvine Elemental |              |        0.423421 |                  |
+| Living Graffiti     |              |                 |         0.385741 |
+| Crystalwing Drake   |              |                 |         0.385211 |
+
+What we have here:
+
+1. Relevance by attribute:
+   - "Look": Three monsters were retrieved (Whispering Shade, Sandstorm Djinn, and Luminoth).
+   - "Habitat": Only one monster from the "Look" results was relevant (Whispering Shade).
+   - "Behavior": Only one monster from the "Look" results was relevant (Luminoth), but it's different from the one relevant for "Habitat".
+
+2. Overall relevance:
+   - No single monster was retrieved for all three attributes simultaneously.
+   - The results are fragmented, with different monsters being relevant for different attributes.
+
+This fragmentation in the results highlights a potential issue with our current search approach, as it fails to find monsters that satisfy all criteria at once.
+
+Maybe we can fix this issue by proactively retrieving more monsters for each attribute?
+Let's try it with 6 monsters for each attribute instead of 3.
+
+| id                  |   score_look |   score_habitat |   score_behavior |
+|:--------------------|-------------:|----------------:|-----------------:|
+| Whispering Shade    |     0.503578 |        0.609567 |                  |
+| Sandstorm Djinn     |     0.407344 |        0.365061 |                  |
+| Luminoth            |     0.378619 |                 |         0.345566 |
+| Nebula Jellyfish    |     0.36627  |                 |         0.259969 |
+| Dreamweaver Octopus |     0.315679 |                 |                  |
+| Quantum Firefly     |     0.288578 |                 |                  |
+| Fungal Network      |              |        0.438856 |                  |
+| Thornvine Elemental |              |        0.423421 |                  |
+| Mist Phantom        |              |        0.366816 |         0.236649 |
+| Stoneheart Golem    |              |        0.342287 |                  |
+| Living Graffiti     |              |                 |         0.385741 |
+| Crystalwing Drake   |              |                 |         0.385211 |
+| Aqua Wraith         |              |                 |         0.283581 |
+
+In total, we retrieved 13 monsters (that is more than half of our tiny dataset!) and still have the same issue: no monster was retrieved for all three attributes simultaneously.
+We could try to increase the number of monsters further, but it would result in the following shortcomings:
+1. While not a problem for our toy task, in production it will result in noticeably longer search time.
+2. With more attributes, the number of monsters to retrieve for every attribute will grow exponentially.
+3. And there is still no guarantee that this will result in retrieval of monsters that satisfy all criteria at once.
+
+This example is enough to show that the naive approach is not suitable for our task.
```

Comment:
I'd add here 2 more general arguments:
- reconciliation of results needs an additional step in a production system vs just return knn results
- multiple knns take more time than 1 knn on larger vectors.

Reply:
or maybe somewhere else in the article? I'd love it if we could pose the full argument in this piece